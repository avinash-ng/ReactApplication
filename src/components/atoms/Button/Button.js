import React from 'react'
import PropTypes from 'prop-types'
const Button = (props ) => {
    return (
        <button className="a_button" className={props.className} onClick={props.onClicked}>{props.buttonName}</button>
    )
}

Button.propTypes = {
    buttonName: PropTypes.string.isRequired
}

export default Button
