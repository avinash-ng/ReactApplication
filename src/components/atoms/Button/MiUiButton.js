import React from 'react'
import { Button } from "@material-ui/core"
const MiUiButton = props => {
    return (
        <Button className={props.className} style={props.style} variant={props.variant} onClick={props.onClick}>{props.buttonName}</Button> 
    )
}
export default MiUiButton