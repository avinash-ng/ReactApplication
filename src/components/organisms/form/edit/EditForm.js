import React, {useState} from 'react'
import InputField from "../../../molecules/Fields/InputField"
import HTMLButton from '../../../atoms/Button/Button'
import axios from 'axios'
// import './editform.css'
import {makeStyles} from '@material-ui/core/styles'
export var confirmEditForm = false


const useStyles = makeStyles((theme) => ({
    mainDiv: {
        textAlign:'center',
        fontSize: '20px',
        marginTop: '100px',
    },
    button: {
        fontSize: '20px',
        marginLeft: '-20px',
        backgroundColor: 'black',
        color: 'white',
        width: '100px',
        height: '40px'
    },
    input: {
        height: '30px',
        width: '250px',
    },
    span: {
        justifyContent: 'center',
        marginRight: '30px'
    }

}))
const EditForm = () => {
    const classes = useStyles()
    const [fileToEdit, updateFileToEdit] = useState('')
    const [newFileName, updateNewFileName] = useState('')
    
    // const onChange = (e) => {
    //     setFile(e.target.files[0])
    // }
    const confirmEdit = () => {
        confirmEditForm = true
        console.log("Confirmed changes....")
        axios.patch("http://localhost:3001/image", {
            oldFileName: fileToEdit,
            newFileName: newFileName,
        }, )
    }

    const onChangeHandlerOldFile = (e) => {
        updateFileToEdit(e.target.value)
    }

    const onChangeHandlerNewFile = (e) => {
        updateNewFileName(e.target.value)
    }

    return (
        <div className="edit-main-div" className={classes.mainDiv}>
        
        <form className="editform">
                
                <h1>File Edit</h1>
                <InputField className={classes.input} label="Old File Name : " type="text" name="old File" value={fileToEdit} onChange={(e)=>onChangeHandlerOldFile(e)}> </InputField> 
                <br /><br/>
               
                <InputField className={classes.input}  label="New File Name :  " type="text" name="new File" value={newFileName} onChange={(e) =>onChangeHandlerNewFile(e) }></InputField><br/><br/>
               
                <HTMLButton className={classes.button} buttonName="Edit" onClicked={confirmEdit}></HTMLButton>
                
            </form>
        </div>
    )
    
}

export default EditForm;
