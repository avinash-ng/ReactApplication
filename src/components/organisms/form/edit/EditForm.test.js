const { shallow, configure, mount } = require("enzyme")
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import Button from "../../../atoms/Button/Button";
import sinon from 'sinon'
import { render, fireEvent } from "@testing-library/react";
import EditForm from './EditForm'
import { confirmEditForm } from './EditForm'
configure({ adapter: new Adapter() });
describe('testing <EditForm /> ', () => {
    it('renders <EditForm /> correctly', () => {
        const RenderComponent = shallow(<EditForm> </EditForm>)
        expect(RenderComponent.exists()).toBe(true)
    })

    it('form submitting functionality', () => {
        const deleteFunction = sinon.spy()
        const RenderComponent = mount(<EditForm > </EditForm>)

        RenderComponent.find(Button).simulate('click')

        expect(confirmEditForm).toBe(true)


    })

    it('matches snapshot', () => {
        const RenderComponent = shallow(<EditForm></EditForm>)
        expect(RenderComponent).toMatchSnapshot()
    })
})