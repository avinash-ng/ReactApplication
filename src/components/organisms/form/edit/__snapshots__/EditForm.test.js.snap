// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <EditForm />  matches snapshot 1`] = `ShallowWrapper {}`;
